### PR TITLE
add map matching API to docs

### PR DESCRIPTION
**Pull request by palmakkar**
Adding new map matching API to docs

### CODE REVIEW

File: docs/api.mdx [hunk 1]
```diff
@@ -1049,6 +1049,90 @@ curl "https://api.radar.io/v1/route/matrix?origins=40.78382,-73.97536&destinatio
...
+
+###### Default rate limit
+
+10 requests per second
+
+###### Sample request
+
+```bash
+curl "https://api.radar.io/v1/route/match?path=440.73421259605101,-73.98635368448336|40.73615136186456,-73.98534685613593|40.73616535996882,-73.98026652961276|40.73701616007588,-73.98238510730718&mode=car&roadAttributes=names, speedLimit, roadClass" \
+  -H "Authorization: prj_live_pk_..."
```
Comment (kochis) [inline docs/api.mdx:1078]: I got an error when running this command:
```js
{"meta":{"code":400,"message":"path: Invalid latitude in 440.73421259605101,-73.98635368448336. Valid range: [-90, 90].","param":"path"}}
```
Also, I think the spaces between commas need to be omitted

Comment (kochis) [reply docs/api.mdx:1078]: I think that first value should be `40.`

File: docs/maps/routing.mdx [hunk 2]
```diff
@@ -34,6 +34,12 @@ Use cases for this endpoint include calculating the walking or driving distance
 
 See the matrix API reference [here](/documentation/api#matrix).
 
+### Map matching
+
+Use the map matching API to snap points collected along a route to roads that were most likely traveled on. Returns a clean path and matched road information.
+
+Use cases for this endpoint include producing clean paths that can be displayed on a map, used for accurate mileage tracking and other analysis, as well as retrieving road context such as speed limits and road names along a traveled path.
+
```
Comment (tjulien) [inline docs/maps/routing.mdx:42]:
```suggestion
See the matching API reference [here](/documentation/api#matching).
```

File: docs/api.mdx [hunk 3]
```diff
@@ -1049,6 +1049,90 @@ curl "https://api.radar.io/v1/route/matrix?origins=40.78382,-73.97536&destinatio
...
+      "longitude": -73.994474,
+      "originalIndex": 2
+    }
+  ],
+  "roadAttributes": [
+    {
+      "names": [
+        "Broadway"
+      ],
+      "speedLimit": 40,
```
Comment (corypisano) [inline docs/api.mdx:1110]: speed limits were updated to an object with `value` and `text` to match Distance API
```
"speedLimit": {
   "value": 40,
   "text": "40 kph"
}
```

File: docs/api.mdx [hunk 4]
```diff
@@ -1049,6 +1049,90 @@ curl "https://api.radar.io/v1/route/matrix?origins=40.78382,-73.97536&destinatio
...
+      "roadClass": "secondary",
+      "originalIndex": 2
+    }
+  ],
+  "geometry": {
+      "polyline": "uafulAln`clCpKzAhCFqAbEs\\heAaA|CuAlEgj@teBm@jBs@|BbBjA~XzR",
+  }
+```
```
Comment (corypisano) [inline docs/api.mdx:1134]: need 1 final `}` at the end. (not indented)

**Review by tjulien — APPROVED** [review]

File: docs/api.mdx [hunk 5]
```diff
@@ -1049,6 +1049,100 @@ curl "https://api.radar.io/v1/route/matrix?origins=40.78382,-73.97536&destinatio
...
+Publishable
+
+###### Default rate limit
+
+10 requests per second
+
+###### Sample request
+
+```bash
+curl "https://api.radar.io/v1/route/match?path=40.73421259605101,-73.98635368448336|40.73615136186456,-73.98534685613593|40.73616535996882,-73.98026652961276|40.73701616007588,-73.98238510730718&mode=car&roadAttributes=names, speedLimit, roadClass" \
```
Comment (jsani-radar) [inline docs/api.mdx:1077]: nit: are there meant to be spaces between these values for `roadAttributes`? or should they be URL encoded as %20? @palmakkar

sorry for the late comment

**Pull request merged**